nilearn/glm/tests/test_first_level.py

```diff
@@ -720,6 +725,14 @@ def test_first_level_from_bids():
                 bids_path, 'main', 'T1w')  # desc not specified
 
 
+def test_first_level_from_bids_with_subject_labels():
+    with InTemporaryDirectory():
+        bids_path = create_fake_bids_dataset(n_sub=10, n_ses=2,
+                                             tasks=['localizer', 'main'],
+                                             n_runs=[1, 3])
+        first_level_from_bids(bids_path, 'main', sub_labels=["foo", "bar"])
```

Comment:
Could the test be more than a smoke test ?

Reply:
+1

You can more explicitly test both conditions under the if else statement. If the path exists you can see if the strings in `sub_labels_exist` and `sub_folders` are equal to a specific string (or more simply, have a length equal to the number of paths that do exist). 

The else statement is tested here but you can add 
```python
with pytest.warns(UserWarning, match=warning_message):
```